code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -303,14 +292,50 @@
 
 /mob/living/simple_animal/hostile/rat/AttackingTarget()
 	. = ..()
-	if(istype(target, /obj/item/food/cheesewedge))
+	if(istype(target, /obj/item/food/cheese))
 		if (health >= maxHealth)
 			to_chat(src, "<span class='warning'>You feel fine, no need to eat anything!</span>")
 			return
 		to_chat(src, "<span class='green'>You eat \the [src], restoring some health.</span>")
-		heal_bodypart_damage(MINOR_HEAL)
+		heal_bodypart_damage(maxHealth)
 		qdel(target)
 
-#undef MINOR_HEAL
-#undef MEDIUM_HEAL
-#undef MAJOR_HEAL
+
+/**
+ *Spittle; harmless reagent that is added by rat king, and makes you disgusted.
+ */
+
+/datum/reagent/rat_spit
+	name = "Rat Spit"
+	description = "Something coming from a rat. Dear god! Who knows where it's been!"
+	reagent_state = LIQUID
+	color = "#C8C8C8"
+	metabolization_rate = 0.03 * REAGENTS_METABOLISM
+	taste_description = "something funny"
+	overdose_threshold = 20
+
+/datum/reagent/rat_spit/on_mob_metabolize(mob/living/L)
+	..()
+	if(HAS_TRAIT(L, TRAIT_AGEUSIA))
+		return
+	to_chat(L, "<span class='notice'>This food has a funny taste!</span>")
+	
+/datum/reagent/rat_spit/overdose_start(mob/living/M)
+	..()
+	var/mob/living/carbon/C = M
```

Comment:
Better variable name, and I don't just mean `carbon`. What is the carbon? The user, the victim, etc.